core/drivers/fsl_sec/hw_key_blob.c

```diff
@@ -235,3 +235,86 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 
 	return res;
 }
+
+#ifdef CFG_RPMB_KEY_WRITE_LOCK
+
+#define RPMB_KEY_WRITE_LOCK_FUSE_BITS 0x80000000
+#define RPMB_KEY_WRITE_LOCK_FUSE_WORD OCOTP_GP1_OFFSET
+
+#ifdef CFG_MX6
+// i.MX6 OCOTP is in CCM_CCGR2
+#define OCOTP_CCG_ENABLE 0x3000
+#elif defined(CFG_MX7)
+// i.MX7 OCOTP is in CCM_CCGR35 (0x4230 offset)
+#endif
+
+TEE_Result tee_otp_check_rpmb_key_write_lock(void)
+{
+	TEE_Result res;
+	uint32_t val;
+
+	res = TEE_SUCCESS;
+
+	DMSG("RPMB INIT: Checking RPMB key write lock");
+
+	val = sec_in32(ptov(OCOTP_BASE + RPMB_KEY_WRITE_LOCK_FUSE_WORD));
+	if (val & RPMB_KEY_WRITE_LOCK_FUSE_BITS) {
+		EMSG("Writing RPMB key forbidden! Write lock fuse is blown!");
+		res = TEE_ERROR_BAD_STATE;
+	}
+
+	return res;
+}
+
+TEE_Result tee_otp_set_rpmb_key_write_lock(void)
+{
+	TEE_Result res;
+	uint32_t val;
+	uint32_t ccg_orig;
+
+	res = TEE_SUCCESS;
+
+	DMSG("RPMB INIT: Setting RPMB key write lock");
+
+	// Enable clock to OTP fuse controller
+	ccg_orig = sec_in32(ptov(CCM_BASE + MX6Q_CCM_CCGR2));
+	sec_out32(ptov(CCM_BASE + MX6Q_CCM_CCGR2), ccg_orig | OCOTP_CCG_ENABLE);
+
+#ifdef CFG_MX6
+	// 0x3E770000 is the OTP unlock key
+	// 0x26 is Bank 4 word 6, 0x27 is Bank 4 Word 7
+#if RPMB_KEY_WRITE_LOCK_FUSE_WORD == OCOTP_GP1_OFFSET
+	val = 0x3E770026;
+#elif RPMB_KEY_WRITE_LOCK_FUSE_WORD == OCOTP_GP2_OFFSET
+abcd
```

Comment:
Left over "is this actually compiling?" junk?

Reply:
Indeed it is.